index.ts

```diff
@@ -29,55 +50,78 @@ const octokit = new Octokit({ auth: process.env.DEVPOOL_GITHUB_API_TOKEN });
-                if (devpoolIssue) {
-                    // if title or state or pricing was updated then update a devpool issue
-                    if (devpoolIssue.title !== projectIssue.title || devpoolIssue.state !== projectIssue.state || getIssuePriceLabel(devpoolIssue) !== getIssuePriceLabel(projectIssue)) {
-                        await octokit.rest.issues.update({
-                            owner: DEVPOOL_OWNER_NAME,
-                            repo: DEVPOOL_REPO_NAME,
-                            issue_number: devpoolIssue.number,
-                            title: projectIssue.title,
-                            state: projectIssue.state,
-                            labels: getDevpoolIssueLabels(projectIssue, ownerName, repoName),
-                        });
-                        console.log(`Updated: ${projectIssue.html_url}`);
-                    } else {
-                        console.log(`No updates: ${projectIssue.html_url}`);
-                    }
-                } else {
-                    // issue does not exist in devpool
-                    // if issue is "closed" then skip it, no need to copy/paste already "closed" issues
-                    if (projectIssue.state === 'closed') continue;
-                    // create a new issue
-                    const createdIssue = await octokit.rest.issues.create({
-                        owner: DEVPOOL_OWNER_NAME,
-                        repo: DEVPOOL_REPO_NAME,
-                        title: projectIssue.title,
-                        body: projectIssue.html_url,
-                        labels: getDevpoolIssueLabels(projectIssue, ownerName, repoName),
-                    });
+  try {
+    // get devpool issues
+    const devpoolIssues: Issue[] = await getAllIssues(
+      DEVPOOL_OWNER_NAME,
+      DEVPOOL_REPO_NAME
+    );
 
-                    console.log(`Created: ${projectIssue.html_url}`);
-                }
-            }
+    // for each project URL
+    for (let projectUrl of projects.urls) {
+      // get owner and repository names from project URL
+      const [ownerName, repoName] = getRepoCredentials(projectUrl);
+      // get all project issues (opened and closed)
+      const projectIssues: Issue[] = await getAllIssues(ownerName, repoName);
+      // for all issues
+      for (let projectIssue of projectIssues) {
+        // if issue exists in devpool
+        const devpoolIssue = getIssueByLabel(
+          devpoolIssues,
+          `id: ${projectIssue.node_id}`
+        );
+        if (devpoolIssue) {
+          const additionalLabelsToAdd = projectIssue?.assignee?.login
+            ? ["Unavailable"]
+            : [];
+          // if title or state or pricing was updated then update a devpool issue
+          if (
+            devpoolIssue.title !== projectIssue.title ||
+            devpoolIssue.state !== projectIssue.state ||
+            devpoolIssue.assignee?.login !== projectIssue?.assignee?.login ||
```

Comment:
This is really hard to review when the whole code is formatted

The `Unavailable` label won't be removed if the partner project's issue assignee is unassigned from the issue because in this case `devpoolIssue.assignee?.login !== projectIssue?.assignee?.login` returns `false`, pls fix

Reply:
I think formatting was necessary so I lint formatted it ! And I also have added the condition where unassigned will be fixed

Reply:
Apologies for the formatting removed it and also added this condition !